### PR TITLE
Add occlusion enabled option to TileMapLayer

### PR DESCRIPTION
**Pull request by sorascode**
Partially Addreses
- https://github.com/godotengine/godot-proposals/issues/10121

Now light occlusion can be enabled/disabled per TileMapLayer

### CODE REVIEW

**Comment by groud** [comment]
Hi, thanks for the contribution. While the change is not that much code, the TileMap node is deprecated. IMHO, unless there's really high demand for it, I would rather avoid new features there.

File: scene/2d/tile_map_layer.cpp [hunk 1]
```diff
@@ -1824,6 +1824,9 @@ void TileMapLayer::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_collision_visibility_mode", "visibility_mode"), &TileMapLayer::set_collision_visibility_mode);
 	ClassDB::bind_method(D_METHOD("get_collision_visibility_mode"), &TileMapLayer::get_collision_visibility_mode);
 
+	ClassDB::bind_method(D_METHOD("set_occlusion_enabled", "enabled"), &TileMapLayer::set_occlusion_enabled);
+	ClassDB::bind_method(D_METHOD("is_occlusion_enabled"), &TileMapLayer::is_occlusion_enabled);
+
```
[the remark this answers is not in the document]
Comment (MarblesFr) [reply scene/2d/tile_map_layer.cpp:1829]: fixed

**Comment by MarblesFr** [comment]
@groud Hi, I exposed the occlusion enabled property in the TileMapLayer node now :) Do you think we can get this merged if I remove the changes in the TileMap node again?

File: scene/2d/tile_map.h [hunk 2]
```diff
@@ -143,6 +143,10 @@ class TileMap : public Node2D {
 	int get_layer_y_sort_origin(int p_layer) const;
 	void set_layer_z_index(int p_layer, int p_z_index);
 	int get_layer_z_index(int p_layer) const;
+	void set_layer_collision_enabled(int p_layer, bool p_enabled);
+	bool is_layer_collision_enabled(int p_layer) const;
```
Comment (AThousandShips) [inline scene/2d/tile_map.h:147]: Regardless if we add the occlusion feature to `TileMap` this code is unrelated and shouldn't be in this PR IMO

**Comment by Mickeon** [comment]
We are in feature freeze so this won't make the cut in 4.3, at earliest in 4.4.

Also do not forget to squash your commits! See the [pull request workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).

**Comment by groud** [comment]
> Do you think we can get this merged if I remove the changes in the TileMap node again?

Yeah, I think it would be good to merge without the changes to the TileMap (in 4.4 though, as, like Mickeon said, we are in feature-freeze for 4.3 now).

**Comment by AThousandShips** [comment]
We can always add the changes to `TileMap` separately if they are desired, along with the collision enable which

**Comment by MarblesFr** [comment]
Alright, thanks for the feedback, the TileMap changes have been removed!

**Review by groud — APPROVED** [review]
Code looks good to me. It should be mergeable for 4.4.
Thanks!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by KoBeWi** [comment]
It's in 4.4 only
